src/main/java/com/powsybl/openloadflow/network/NominalVoltageMapping.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.network;
+
+import com.powsybl.iidm.network.Bus;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.Terminal;
+import org.apache.commons.lang3.mutable.MutableInt;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * Nominal voltage mapping. All nominal voltages that are closed (10% threshold) are merged together so that we only
+ * keep ones containing the biggest number of buses, resulting a set of common nominal voltages and discarding unusual
+ * and rare ones. The purpose of this is to avoid the addition a voltage ratio to line with a small difference of
+ * nominal voltage between side 1 and 2.
+ *
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class NominalVoltageMapping {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NominalVoltageMapping.class);
+
+    public static final NominalVoltageMapping NONE = new NominalVoltageMapping();
+
+    private final Map<Double, Double> mapping;
+
+    public NominalVoltageMapping() {
+        this(new HashMap<>());
+    }
+
+    public NominalVoltageMapping(Map<Double, Double> mapping) {
+        this.mapping = Objects.requireNonNull(mapping);
+    }
+
+    public double get(double nominalV) {
+        return mapping.getOrDefault(nominalV, nominalV);
+    }
+
+    public double get(Terminal terminal) {
+        return get(Objects.requireNonNull(terminal).getVoltageLevel().getNominalV());
+    }
+
+    public double get(Bus bus) {
+        return get(Objects.requireNonNull(bus).getVoltageLevel().getNominalV());
+    }
+
+    private static Map<Double, MutableInt> createHistogram(Network network) {
+        Map<Double, MutableInt> nominalVoltageHistogram = new TreeMap<>();
+        for (var vl : network.getVoltageLevels()) {
+            nominalVoltageHistogram.computeIfAbsent(vl.getNominalV(), k -> new MutableInt())
+                    .increment();
+        }
+        return nominalVoltageHistogram;
+    }
+
+    public static NominalVoltageMapping create(Network network, double resolution) {
```

Comment:
The only thing I wonder is if we could need different "strategies" to decide on the mapping:
for example should we take into account connectivity, so that a nominal voltage in Portugal does not affect the per-unit in Germany.
In that case we could make it an interface with other impls in the future. But this can be done when it will really be needed, in any case.

Reply:
I tried to do that first, but the problem I had is that I need to map new nominal voltages on bus ids (we cannot associate it to voltage level id) and for branches connected to only one side, if side 2 is open I cannot get nominal voltage to per unit  the branch. Any idea if there is a simple way to overcome this?

Reply:
Indeed I didn't think of those "details", it gets tricky ... even more if the topology changes during computation, so it does not seem such a good idea after all !

Reply:
I refactored to have a generic  interface and a simple implementation so that if one day we want to have a more complex implementations, we can dot it without changing to much code